include/xtl/xvariant_impl.hpp

```diff
@@ -2660,20 +2660,20 @@ namespace mpark {
 #ifdef MPARK_CPP14_CONSTEXPR
   namespace detail {
 
-    inline constexpr bool all(std::initializer_list<bool> bs) {
```

Comment:
It is not recommended to modify this file since it is a copy/paste from https://github.com/mpark/variant that we try to maintain up to date. Therefore the next time we update it, these changes will be lost. It would be better to open a PR upstream and then open a PR here with the replacement for this file.

Out of curiosity, do you know why the parameter pack was not expanded with the previous implementation?

Reply:
Well, the nvidia compiler seems to screw up when having the not operator before the parameter pack. I tried to isolate this in a minimal program:

```c++
#include <iostream>
#include <initializer_list>

inline constexpr bool all(std::initializer_list<bool> bs) {
  for (bool b : bs) {
    if (!b) {
      return false;
    }
  }
  return true;
}

template<typename... Vs>
inline constexpr decltype(auto) visit(Vs &&... vs) {
  return all({!vs.thing()...}) ? (void)0 :
     void(0);
}

struct FalseThing
{
  inline bool thing() { return false; }
};
struct TrueThing
{
  inline bool thing() { return true; }
};

int main()
{
  FalseThing a, b;
  TrueThing c, d;
  visit(a, c, b, d);
}
```

But this compiles just fine, so it's very hard for me to see where the nvcc parsing screws up.

Reply:
OK, so, I made that PR upstream then!

Reply:
And, as a small update, seems that this is going to be resolved in mpark/variant some time soon. I'll change this PR accordingly to simply adopt the new mpark/variant once that's said and done.

Reply:
OK, the changes to variant.hpp have been removed!